src/optimal_image/search.rs

```diff
@@ -0,0 +1,27 @@
+use std::cmp::Ordering;
+use ImageConfig;
+use ImageScore;
+
+pub fn find_optimal_config(config: &Vec<ImageConfig>, threshold: f64) -> &ImageConfig {
+    let index = config.binary_search_by(|probe| {
```

Comment:
binary_search_by is pretty cool, it basically let you write your own visitor for the binary search over collections, this way to find the best config between 70 and 90 it will take max 5 steps being sure to not leave any possibility out